### PR TITLE
Feature/add poses to flow

### PR DESCRIPTION
**Pull request by CathrinTruchan**
I added a functionality with which you can compose your individual flow from a selection of asanas. 
For this I added a scrollable list to the flow page and two states to render the selected poses. 
Here is the link to the preview: 
https://capstone-project-16uwigjs3-cathrintruchan.vercel.app/flow

### CODE REVIEW

File: pages/_app.js [hunk 1]
```diff
@@ -1,10 +1,19 @@
...
+import { useState } from "react";
 
 function MyApp({ Component, pageProps }) {
+  const [flowAsanaIds, setFlowAsanaIds] = useState([]);
+  const [selectedAsanaIds, setSelectedAsanaIds] = useState([]);
   return (
     <>
       <GlobalStyle />
-      <Component {...pageProps} />
+      <Component
```
Comment (Ozanyurdakul98) [inline pages/_app.js:10]: Hey Cathrin! off course you can do it like this. But maybe you would like to consider wrapping your  <Component {...pageProps} around a Layout Component and put your StateUpLiftings in there. Would look more cleaner I think

Comment (Ozanyurdakul98) [reply pages/_app.js:10]: But not required

File: pages/flow/index.js [hunk 2]
```diff
@@ -1,22 +1,48 @@
...
   });
 
   return (
     <>
-      <h2>Your flow for today:</h2>
+      <StyledH2>Choose your Asanas for today:</StyledH2>
+      <AddAsanaSection>
+        <StyledList>
+          {asanas.map((asana) => {
+            return (
```
Comment (Ozanyurdakul98) [inline pages/flow/index.js:22]: Ich glaube du brauchst hier gar kein return. Check das mal bitte. Es gab bei einer Ähnlichen Konstellation bei mir nämlich ab und zu dann Crashs.

**Review by Ozanyurdakul98 — APPROVED** [review]
Hey, just 1 or 2 advises, but everything looks right/correct

File: pages/flow/index.js [hunk 3]
```diff
@@ -33,3 +59,41 @@ export default function Flow() {
 const StyledList = styled.ul`
   list-style: none;
 `;
+
+const StyledListItem = styled.li`
+  display: flex;
```
Comment (kai-franke) [inline pages/flow/index.js:64]: Only for UX: Did you try adding something like `justify-content: space-between;`?
Bildschirmfoto 2022-10-19 um 18 34 52

File: pages/flow/index.js (hunk 2 above)
Comment (kai-franke) [reply pages/flow/index.js:22]: Yes, this should work as an implicit return as well.

**Review by kai-franke — APPROVED** [review]
Really great! I have just some minor remarks.

File: pages/flow/index.js (hunk 2 above)
Comment (CathrinTruchan) [reply pages/flow/index.js:22]: Du hast Recht. Danke für den Tip!

File: pages/flow/index.js (hunk 3 above)
Comment (CathrinTruchan) [reply pages/flow/index.js:64]: Thanks for the hint! UX ist not great yet. Will work on that later (hopefully). But with space-between it already looks better :)

File: pages/_app.js (hunk 1 above)
Comment (CathrinTruchan) [reply pages/_app.js:10]: Thanks for the hint. I will add a Layout Component in a later user story (when I add navigation etc). I will try to put the states in there then.

**Comment by CathrinTruchan** [comment]
> Hey, just 1 or 2 advises, but everything looks right/correct

Thanks very much!!

**Comment by CathrinTruchan** [comment]
> Really great! I have just some minor remarks.

Thanks a lot!!

File: pages/flow/index.js [hunk 4]
```diff
@@ -1,35 +1,156 @@
...
 
-  const flowAsanas = flowAsanaIds.map((item) => {
-    return getAsanaByID(item);
+  function sumUpAsanas() {
+    setCountAsanas((prev) => prev + 1);
+  }
+
+  const flowAsanas = flowAsanaIds.map((id) => {
+    return getAsanaByID(id);
   });
```
Comment (Roland-Hufnagel) [inline pages/flow/index.js:18]: Ist das nicht ein bisschen doppelt? Warum speicherst du die Asanas nicht gleich im useState?

File: pages/flow/index.js [hunk 5]
```diff
@@ -1,35 +1,156 @@
...
+  border-radius: 22px;
+  text-align: center;
+`;
+
+const SectionHeader = styled.section`
+  display: flex;
+  justify-content: space-between;
+  margin-bottom: 2rem;
+  padding-bottom: 1rem;
+  border-bottom: 1px solid grey;
```
Comment (philmetscher) [inline pages/flow/index.js:155]: never use colors like `red`. Use Hex-Colors or RGB or HSL.
Colors like `red` can be interpreted differently by different browsers

Pls look inside all styled Components and correct this

File: pages/flow/index.js [hunk 6]
```diff
@@ -1,35 +1,156 @@
...
+
+const CloseButton = styled.div`
+  background-color: #5d6bea;
+  color: #f5f5f5;
+  padding: 0;
+  width: 1.5rem;
+  height: 1.5rem;
+  border-radius: 50%;
+  text-align: center;
+  cursor: pointer; ;
```
Comment (philmetscher) [inline pages/flow/index.js:137]: two semicolons?

File: pages/flow/index.js [hunk 7]
```diff
@@ -1,35 +1,156 @@
...
+            {asanas.map((asana) => (
+              <StyledListItem key={asana.id}>
+                <h3>{asana.english_name}</h3>
+                <StyledAddButton
+                  onClick={() => {
+                    setFlowAsanaIds([...flowAsanaIds, asana.id]);
+                    sumUpAsanas();
+                  }}
+                >
+                  <a>+</a>
```
Comment (Roland-Hufnagel) [inline pages/flow/index.js:58]: Ein Anchor??? Warum?

**Review by philmetscher — CHANGES_REQUESTED** [review]
I added some comments :)
You should add these css statement before you start a project
```css
* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}
html {
  font-size: 100%;
}
```

File: pages/flow/index.js (hunk 4 above)
Comment (CathrinTruchan) [reply pages/flow/index.js:18]: Ich prüfe wie besprochen nochmal, wie sich das später mit dem local storage verhält.

File: pages/flow/index.js (hunk 7 above)
Comment (CathrinTruchan) [reply pages/flow/index.js:58]: Gute Frage ;) Nehm ich raus.

File: pages/flow/index.js [hunk 8]
```diff
@@ -42,16 +27,31 @@ export default function Flow({
                     setSelectedAsanaIds([...selectedAsanaIds, asana.id])
                   }
                 >
-                  Add to flow
+                  <a>Add to flow</a>
```
[the remark this answers is not in the document]
Comment (CathrinTruchan) [reply pages/flow/index.js]: That was an older version. I have already removed it. But thanks for the hint!

File: pages/flow/index.js [hunk 9]
```diff
@@ -1,35 +1,167 @@
...
 
-  const flowAsanas = flowAsanaIds.map((item) => {
-    return getAsanaByID(item);
+  function sumUpAsanas() {
+    setCountAsanas((prev) => prev + 1);
+  }
+
+  function autoScroll() {
+    setTimeout(() => {
+      window.scrollBy({ top: 500, behavior: "smooth" });
```
Comment (kai-franke) [inline pages/flow/index.js:18]: Awesome! 😍

File: pages/flow/index.js [hunk 10]
```diff
@@ -1,35 +1,167 @@
...
+const StyledH2 = styled.h2`
+  text-align: center;
+`;
+
+const StyledButton = styled.button`
+  width: 16rem;
+  margin: -15rem auto 5rem auto;
+  display: block;
+  border: none;
+  background-color: #ff8667;
```
Comment (philmetscher) [inline pages/flow/index.js:128]: try to store all the color variables in your `:root`. It is easier to change everything if you don't like the color.

**Review by philmetscher — APPROVED** [review]

**Review by kai-franke — APPROVED** [review]

File: pages/flow/index.js (hunk 10 above)
Comment (CathrinTruchan) [reply pages/flow/index.js:128]: Yes, you are right ;) I will refactor that. Thanks for the review!

**Pull request merged**